TeamCode/src/main/java/org/firstinspires/ftc/teamcode/opmodes/auto/CompetitionAutonomous.java

```diff
@@ -29,15 +29,29 @@ public enum From {
         START, CONE_STACK
     }
 
+    public enum TeamState {
+        RED_RIGHT,
+        RED_LEFT,
+        BLUE_RIGHT,
+        BLUE_LEFT
+    }
+
     private From startPosition;
     private State mCurrentState;                         // Current State Machine State.
-
+    protected TeamState teamState;
+    private int sign;
 
     /**
      * Initializes State Machine
      */
-    public void init() {
+    public void init(TeamState teamState) {
```

Comment:
Change params to `color` and `side` and (create and) set two fields to keep track of them